app/models/piece.rb

```diff
@@ -1,6 +1,79 @@
 class Piece < ApplicationRecord
   belongs_to :game
 
+  def is_obstructed?(game, pos2)
+    
+    @occupied_positions =[]
```

Comment:
no need for '@'. its used to declare instance variables... in this case you just use it as a local variable inside your method